packages/kbn-esql-validation-autocomplete/src/definitions/types.ts

```diff
@@ -8,6 +8,24 @@
 
 import type { ESQLCommand, ESQLCommandOption, ESQLFunction, ESQLMessage } from '@kbn/esql-ast';
 
+export type ParameterType =
```

Comment:
Strict typing on these will give us greater confidence, especially as we start using the generated definitions in https://github.com/elastic/kibana/issues/179634

Reply:
Also, when we do this: https://github.com/elastic/kibana/issues/174710